relay-event-normalization/src/normalize/mod.rs

```diff
@@ -207,6 +210,46 @@ fn normalize_app_start_spans(event: &mut Event) {
     }
 }
 
+/// Replaces all incoming metric identifiers in the metric summary with the correct MRI.
+///
+/// The reasoning behind this normalization, is that the SDK sends namespace-agnostic metric
+/// identifiers in the form `metric_type:metric_name@metric_unit` and those identifiers need to be
+/// converted to MRIs in the form `metric_type:metric_namespace/metric_name@metric_unit`.
+fn normalize_metrics_summary_mris(value: &mut Value) {
+    let Value::Object(metrics) = value else {
+        return;
+    };
+
+    let metrics = mem::take(metrics)
+        .into_iter()
+        .filter_map(|(key, value)| {
+            Some((
+                MetricResourceIdentifier::parse(&key).ok()?.to_string(),
+                value,
+            ))
+        })
+        .collect();
+
+    *value = Value::Object(metrics);
+}
+
+/// Normalizes all the metrics summaries across the event payload.
+fn normalize_all_metrics_summaries(event: &mut Event) {
+    if let Some(metrics_summary) = event._metrics_summary.value_mut().as_mut() {
+        normalize_metrics_summary_mris(metrics_summary)
+    }
+
+    if let Some(spans) = event.spans.value_mut() {
```

Comment:
Early return with `let .. else` would make this less indentation heavy.

```rs
    let Some(spans) = event.spans.value_mut() else {
        return;
    };
```

Could also consider combining the inner `if let`'s:

```rs
    let Some(spans) = event.spans.value_mut() else {
        return;
    };

    for span in spans.iter_mut() {
        let metrics_summary = span
            .value_mut()
            .as_mut()
            .and_then(|span| span._metrics_summary.value_mut().as_mut());

        if let Some(ms) = metrics_summary {
            normalize_metrics_summary_mris(ms)
        }
    }
```